broker/src/main/java/io/zeebe/broker/engine/impl/SnapshotChunkImpl.java

```diff
@@ -91,6 +98,9 @@ public void wrap(final DirectBuffer buffer, final int offset, final int length)
     snapshotId = decoder.snapshotId();
     chunkName = decoder.chunkName();
     checksum = decoder.checksum();
+    if (decoder.sbeSchemaVersion() >= SnapshotChunkDecoder.snapshotChecksumSinceVersion()) {
```

Comment:
What will happen, if sender and receiver has different schema versions?

Reply:
I didn't come up with a way to test this easily but when the receiver is older than the sender it should not even read the snapshot checksum. If it's newer than the sender, then the message's version is checked and the null value is used instead. Actually this check is redundant because it's already done in `snapshotChecksum()`, so I'll remove it